### PR TITLE
feat(utils): add embed builder

### PR DESCRIPTION
**Pull request by MatthewSH**
This is a PR implementing the embed builder from #3124, based on PR #2931.

# Differences from other PR
- This is only embed builders, others will come later.
- Uses DD Discord interfaces and uses `Omit` to pull out any key we use as required.
- Uses [ECMAScript private getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#private_methods) to get the current embed instead of a method.
- Simplified code a bit, the modifying the getter directly vs setting new variable and modifying. This is more preference.
- Using `set` before method names to make it more descriptive of intent of method
- Custom `EmbedBuilderError`
- Use spread operations where possible.
- Optional `validate` method that check all embeds to ensure they're in the [documented embed limits](https://discord.com/developers/docs/resources/channel#embed-object-embed-limits) for added safety if desired.

----

```[tasklist]
# TODO
- [x] Complete `TODO` in code
- [x] Add method documentation
- [x] Finish the `validate` method
- [x] Testing
```

### CODE REVIEW

File: packages/utils/src/builders/embeds.ts [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+}
+
+/**
+ * Custom EmbedsBuilder error.
+ *
+ * @class EmbedsBuilderError
+ * @typedef {EmbedsBuilderError}
+ * @extends {Error}
+ */
+class EmbedsBuilderError extends Error {
```
Comment (Skillz4Killz) [inline packages/utils/src/builders/embeds.ts:381]: Why is this class needed?

Comment (MatthewSH) [reply packages/utils/src/builders/embeds.ts:381]: @Skillz4Killz mainly just experience. Setting a custom error allows you to check, if desired, what the error name is and allows you to handle that error differently than others.

There's no real downside to not throwing generic errors vs custom errors. I don't feel strongly either way, I can just throw a generic error instead if that's what we decide.

Comment (Skillz4Killz) [reply packages/utils/src/builders/embeds.ts:381]: I think we should keep it generic please. You can compare error message as well.

Comment (MatthewSH) [reply packages/utils/src/builders/embeds.ts]: Resolved in 38e1a5ba0678c5db45cb117a605da37903fd6dc1

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**